packages/ibm-products/src/components/OptionsTile/OptionsTile.js

```diff
@@ -1,291 +1,143 @@
+    const handleToggle = (toggled) => {
+      setToggled(toggled);
+      if (onToggle) {
+        onToggle(toggled);
       }
-    }
-
-    function collapse() {
-      if (contentRef.current && !reducedMotion.matches) {
-        setClosing(true);
+    };
 
-        const { paddingTop, paddingBottom, height } = getComputedStyle(
-          contentRef.current
-        );
-
-        const animation = contentRef.current.animate(
-          [
-            {
-              paddingTop,
-              paddingBottom,
-              height,
-              opacity: 1,
-            },
-            {
-              paddingTop: 0,
-              paddingBottom: 0,
-              height: 0,
-              opacity: 0,
-            },
-          ],
-          {
-            duration: Number(carbonMotion.moderate01.replace('ms', '')),
-            easing: carbonMotion.easings.entrance.productive,
-          }
-        );
-
-        const callback = () => {
-          setIsOpen(false);
-          setClosing(false);
-        };
-
-        animation.onfinish = callback;
-        animation.oncancel = callback;
-      } else {
-        // in case the ref is not set or the user prefers reduced motion, skip the animation
-        setIsOpen(false);
+    const getIcon = () => {
+      if (locked) {
+        return Locked;
       }
-    }
-
-    function toggle(e) {
-      e.preventDefault();
-
-      if (isOpen) {
-        collapse();
-      } else {
-        expand();
+      if (invalid) {
+        return WarningFilled;
       }
-    }
-
-    function renderTitle() {
-      let Icon = null;
-      let text = summary;
-      const summaryClasses = [`${blockClass}__summary`];
+      if (warn) {
+        return WarningAltFilled;
+      }
+    };
 
+    const getText = () => {
+      if (locked) {
+        return lockedText;
+      }
       if (invalid) {
-        Icon = WarningFilled;
-        text = invalidText;
-        summaryClasses.push(`${blockClass}__summary--invalid`);
-      } else if (warn) {
-        Icon = WarningAltFilled;
-        text = warnText;
-        summaryClasses.push(`${blockClass}__summary--warn`);
-      } else if (locked) {
-        Icon = Locked;
-        summaryClasses.push(`${blockClass}__summary--locked`);
-
-        if (!text) {
-          text = lockedText;
-        }
+        return invalidText;
       }
-
-      const hasValidationState = invalid || warn || locked;
-      const summaryHidden = !hasValidationState && enabled === false;
-
-      if (summaryHidden) {
-        summaryClasses.push(`${blockClass}__summary--hidden`);
+      if (warn) {
+        return warnText;
       }
+      return summary;
+    };
+
+    const renderTitle = () => {
+      const Icon = getIcon();
+      const text = getText();
 
       return (
         <div className={`${blockClass}__heading`}>
-          <h6 id={titleId} className={`${blockClass}__title`}>
-            {title}
-          </h6>
+          <h6 className={`${blockClass}__title`}>{title}</h6>
```

Comment:
It look like `props.id` is not used anywhere even though the toggle's `aria-labelledby` refers to it.

```suggestion
          <h6 id={id} className={`${blockClass}__title`}>{title}</h6>
```